packages/commands/src/index.ts

```diff
@@ -537,14 +537,36 @@ export class CommandRegistry {
       event
     );
 
-    // If there is no exact match and no partial match, replay
-    // any suppressed events and clear the pending state.
-    if (!exact && !partial) {
-      this._replayKeydownEvents();
+    // Remove modifier key only keystrokes from the current key sequence.
+    // keystrokes that are modifier key(s) plus another key are not affected
+    // intended functionality: Alt then Alt 1 should result to: ['Alt'] then ['Alt 1']
+    // Removing this code results in mod key duplication: ['Alt', 'Alt 1']
+    if (
+      // Check that only a mod key has been pressed
+      (event.altKey && event.key === 'Alt') ||
+      (event.ctrlKey && event.key === 'Ctrl') ||
+      (event.shiftKey && event.key === 'Shift')
+    ) {
+      this._keystrokes.length = 0;
+    }
+
+
+    // If there is an exact match that is not excluded and no partial match, the exact match
+    // can be dispatched immediately. The pending state is cleared so
+    // the next key press starts from the default state.
+    // add new modifier only key commands that should be excluded to the end of the if statement
+    if (
+      exact &&
+      !partial &&
+      !exact?.command.includes('application:activate-sidebar-overlays')
```

Comment:
This is Jupyterlab specific, and should not be used in Lumino.

Reply:
I have generalised it to mod key(s)